synapse/event_auth.py

```diff
@@ -491,7 +494,7 @@ def _is_membership_change_allowed(
         key = (EventTypes.Create, "")
         create = auth_events.get(key)
         if create and event.prev_event_ids()[0] == create.event_id:
-            if create.content["creator"] == event.state_key:
+            if create.sender == event.state_key:
```

Comment:
Sanity check: does today's Synapse validate that a creation event's creator is equal to its sender?

Reply:
Hmmm,.. Synapse *always* overwrites it on creation, but I'm not sure it is validated. Will check.

Reply:
In a confusing way Synapse would reject an invalid creator joining the room (i.e. the `m.room.member` event for the creator wouldn't work) due to:

https://github.com/matrix-org/synapse/blob/a368d30c1cfe7457fca4fcdd03ae481ba65a226c/synapse/event_auth.py#L488-L495

This pretty much says to allow the `m.room.member` event if there's only a single previous event (the create event), the user is joining, and the `creator` field is the member joining.

The [auth rules](https://spec.matrix.org/v1.6/rooms/v10/#authorization-rules) (see 4.3.1) say:

> If the only previous event is an `m.room.create` and the `state_key` is the creator, allow.

No where does it say that the `creator` must be the `sender` explicitly (either in the auth rules or in the code). I suspect this should depend on the room version?